src/platform/node/metrics.js

```diff
@@ -0,0 +1,216 @@
+'use strict'
+
+const v8 = require('v8')
+const log = require('../../log')
+
+const INTERVAL = 10 * 1000
+
+let nativeMetrics = null
+
+let interval
+let client
+let time
+let cpuUsage
+let counters
+
+reset()
+
+module.exports = function () {
+  return {
+    start: () => {
+      const StatsD = require('hot-shots')
+
+      try {
+        nativeMetrics = require('../../../build/Release/metrics')
+        nativeMetrics.start()
+      } catch (e) {
+        log.error('Unable to load native metrics module. Some metrics will not be available.')
+      }
+
+      client = new StatsD({
+        host: this._config.hostname,
+        port: 8125, // TODO: allow to configure this
+        prefix: 'datadog.tracer.runtime.',
+        globalTags: {
+          'env': this._config.env,
+          'service': this._config.service,
+          'runtime-id': this._config.runtimeId,
+          'language': 'javascript',
+          'runtime': 'node'
+        },
+        errorHandler: () => {}
+      })
+
+      time = process.hrtime()
+
+      if (nativeMetrics) {
+        interval = setInterval(() => {
+          captureCommonMetrics()
+          captureNativeMetrics()
+        }, INTERVAL)
+      } else {
+        cpuUsage = process.cpuUsage()
```

Comment:
Do we need to call this before setting interval for `if (nativeMetrics)`?

Reply:
This is not needed since for native metrics it's done in C++.